cmake/modules/FindMayaUsd.cmake

```
@@ -0,0 +1,90 @@
+#
+# Simple module to find MayaUsd.
+#
+# This module searches for a valid MayaUsd installation.
+# It searches for MayaUsd's libraries and include header files.
+#
+# Variables that will be defined:
+# MAYAUSD_FOUND           Defined if a MayaUsd installation has been detected
+# MAYAUSD_LIBRARY_DIR     Path to MayaUsd libraries directory
+# MAYAUSD_INCLUDE_DIR     Path to the MayaUsd include directory
+# MAYAUSDAPI_LIBRARY      Path to MayaUsdAPI library
+# MAYAUSD_MOD_PATH        Path to mayaUSD.mod file
+#
+
+find_path(MAYAUSD_INCLUDE_DIR
+        mayaUsd/mayaUsd.h
+    HINTS
+        ${MAYAUSD_LOCATION}
+    PATH_SUFFIXES
+        include/
+    DOC
+        "MayaUsd header path"
+)
+
+find_path(MAYAUSD_LIBRARY_DIR
+        mayaUsd.lib
+    HINTS
+        ${MAYAUSD_LOCATION}
+    PATH_SUFFIXES
+        lib/
+    DOC
+        "MayaUsd libraries path"
+)
+
+find_library(MAYAUSDAPI_LIBRARY
+    NAMES
+        mayaUsdAPI
+    HINTS
+        ${MAYAUSD_LOCATION}
+    PATHS
+        ${MAYAUSD_LIBRARY_DIR}
+    PATH_SUFFIXES
+        lib/
+    DOC
+        "mayaUsdAPI library"
+    NO_DEFAULT_PATH
+)
+
+find_path(MAYAUSD_MOD_PATH
+        mayaUSD.mod
+    HINTS
+        ${MAYAUSD_LOCATION}
+    DOC
+        "mayaUSD.mod path"
+)
+
+
+# Handle the QUIETLY and REQUIRED arguments and set MAYAUSD_FOUND to TRUE if
+# all listed variables are TRUE.
+include(FindPackageHandleStandardArgs)
+
+find_package_handle_standard_args(MAYAUSD
+    REQUIRED_VARS
+        MAYAUSD_INCLUDE_DIR
+        MAYAUSD_LIBRARY_DIR
+)
+
+if(MAYAUSD_FOUND)
+    message(STATUS "MayaUsd found : ${MAYAUSD_LOCATION}")
+    message(STATUS "MayaUsd include dir: ${MAYAUSD_INCLUDE_DIR}")
+    message(STATUS "MayaUsd libraries dir: ${MAYAUSD_LIBRARY_DIR}")
+    message(STATUS "MayaUsdAPI.lib fullpath : ${MAYAUSDAPI_LIBRARY}")
```

I would not specify .lib - that is Windows specific.
```suggestion
    message(STATUS "MayaUsdAPI library : ${MAYAUSDAPI_LIBRARY}")
```